prover/prover_dal/migrations/20240131134938_initial-prover-migration.up.sql

```diff
@@ -0,0 +1,287 @@
+    updated_at                     TIMESTAMP NOT NULL
+);
+
+CREATE INDEX IF NOT EXISTS idx_scheduler_dependency_tracker_fri_circuit_ids_filtered
+    ON public.scheduler_dependency_tracker_fri (circuit_1_final_prover_job_id, circuit_2_final_prover_job_id, circuit_3_final_prover_job_id, circuit_4_final_prover_job_id, circuit_5_final_prover_job_id, circuit_6_final_prover_job_id, circuit_7_final_prover_job_id, circuit_8_final_prover_job_id, circuit_9_final_prover_job_id, circuit_10_final_prover_job_id, circuit_11_final_prover_job_id, circuit_12_final_prover_job_id, circuit_13_final_prover_job_id)
+    WHERE (status <> 'queued'::text);
+
+---- SCHEDULER WITNESS JOBS FRI
+
+---- Scheduler jobs. Each is picked by a WG which produces input for provers.
+CREATE TABLE IF NOT EXISTS scheduler_witness_jobs_fri (
+    l1_batch_number                  BIGINT NOT NULL PRIMARY KEY,
+    scheduler_partial_input_blob_url TEXT NOT NULL,
+    status                           TEXT NOT NULL,
+    processing_started_at            TIMESTAMP,
+    time_taken                       TIME,
+    error                            TEXT,
+    created_at                       TIMESTAMP NOT NULL,
+    updated_at                       TIMESTAMP NOT NULL,
+    attempts                         SMALLINT DEFAULT 0 NOT NULL,
+    protocol_version                 INTEGER REFERENCES prover_fri_protocol_versions (id),
+    picked_by                        TEXT
+);
+
+CREATE INDEX IF NOT EXISTS idx_scheduler_fri_status_processing_attempts
+    ON public.scheduler_witness_jobs_fri (processing_started_at, attempts)
+    WHERE (status = ANY (ARRAY['in_progress'::text, 'failed'::text]));
+
+---- WITNESS INPUTS FRI
+
+-- Witness input jobs. Each is picked by a WG which produces input for provers.
+CREATE TABLE IF NOT EXISTS witness_inputs_fri (
+    l1_batch_number            BIGINT NOT NULL PRIMARY KEY,
+    merkle_tree_paths_blob_url TEXT,
+    attempts                   SMALLINT DEFAULT 0 NOT NULL,
+    status                     TEXT NOT NULL,
+    error                      TEXT,
+    created_at                 TIMESTAMP NOT NULL,
+    updated_at                 TIMESTAMP NOT NULL,
+    processing_started_at      TIMESTAMP,
+    time_taken                 TIME,
+    -- Note that is_blob_cleaned column is not used, but as part of just mirroring mainnet prover tables, it'll be kept.
+    -- To be deleted with a follow-up migration.
+    is_blob_cleaned            BOOLEAN,
+    protocol_version           INTEGER REFERENCES prover_fri_protocol_versions (id),
+    picked_by                  TEXT
+);
+
+CREATE INDEX IF NOT EXISTS idx_witness_inputs_fri_status_processing_attempts
+    ON witness_inputs_fri (processing_started_at, attempts)
+    WHERE (status = ANY (ARRAY['in_progress'::text, 'failed'::text]));
+CREATE INDEX IF NOT EXISTS idx_witness_inputs_fri_queued_order
+    ON witness_inputs_fri (l1_batch_number)
+    WHERE (status = 'queued'::text);
```

Comment:
very well, separte thanks for documenting this. By the way it's possible to document them in SQL (so that description is not only available on this file, but in the DB itself - Postgres clients will render this description.
Optional.